extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -176,6 +176,14 @@ export class ResourceTypePickerDialog extends DialogBase {
 
 	private selectResourceType(resourceType: ResourceType): void {
 		this._selectedResourceType = resourceType;
+		//handle special case when resource type has different OK button.
+		if (this._selectedResourceType.okButtonText) {
```

Comment:
simplify to:
this._dialogObject.okButton.label = this.selectedResourceType.okButtonText || localize('deploymentDialog.OKButtonText', "Select");

Reply:
See in localizedConstants.ts to see if there is already a constant defined there for "Select". If yes then consider reusing it, else please define this constant there and use it here.